src/Engine/ProtoScript/Runners/LiveRunner.cs

```diff
@@ -254,6 +256,36 @@ private void ApplyChangeSetForceExecute(ChangeSetData changeSet)
             }
         }
 
+        private void ReActivateGraphNodesInCycle(List<AssociativeNode> nodeList)
+        {
+            if (nodeList == null || !nodeList.Any()) return;
+
+            var assocGraph = core.DSExecutable.instrStreamList[0].dependencyGraph;
+            var graphNodes = assocGraph.GetGraphNodesAtScope(Constants.kInvalidIndex, Constants.kInvalidIndex);
+
+            foreach (var node in nodeList)
+            {
+                var bNode = node as BinaryExpressionNode;
+
+                var identifier = bNode?.LeftNode as IdentifierNode;
+                if (identifier == null) continue;
+
+                GraphNode rootNode = null;
+                foreach(var gNode in graphNodes)
+                {
+                    if(identifier.Value == gNode.updateNodeRefList[0].nodeList[0].symbol.name)
+                    {
+                        rootNode = gNode;
```

Comment:
isn't it possible that a graph has two cycles that have been fixed? Won't this only clear the first?

Reply:
Thanks for pointing this out. I will check on this.